docs/draft-alakuijala-brotli-07.nroff

```diff
@@ -1008,9 +1008,10 @@ lengths and distances, respectively, is encoded in the meta-block
 header, and it must equal to the largest block type plus one in that
 block category. In other words, the set of literal, insert-and-copy
 length and distance block types must be [0..NBLTYPESL-1],
-[0..NBLTYPESI-1], and [0..NBLTYPESD-1], respectively. From this it
-follows that the alphabet size of literal, insert-and-copy length and
-distance block type codes is NBLTYPES + 2, NBLTYPESI + 2 and
+[0..NBLTYPESI-1], and [0..NBLTYPESD-1], respectively. If a block type
+has a value outside its respective range, the stream should be rejected as
+invalid. From this it follows that the alphabet size of literal, insert-and-copy
+length and distance block type codes is NBLTYPES + 2, NBLTYPESI + 2 and
```

Comment:
I am starting to wonder if we need this extra sentence here.

Since the block types are read using a prefix code with a certain alphabet size, there is no way we can decode a block type that it outside the range of its alphabet, right?

Was your intention to address the overflowing of the "previous block type +1" symbol? That behavior is already defined above:
"If the current block type is the maximal possible, then a block type symbol of 1 results in wrapping to a new block type of 0."